2020/10/27/craft-a-cms-for-developers.html.md

```diff
@@ -0,0 +1,234 @@
+---
+author: "Kevin Campusano"
+title: "Craft: A CMS for developers"
+tags: craft, cms, php, docker
+---
+
+![Banner](craft-a-cms-for-developers/banner.png)
+
+As a software engineer, I thrive and thoroughly enjoy working on fully custom software products. Applications conceived to model and help in the execution of some business process and that are built from the ground up by a team of developers.
+
+Such projects are often complex and expensive though, and for some clients, they can be overkill. Some clients come up with requirements that are better served by off the shelf software solutions. One group of such solutions are [Content Management Systems](https://en.wikipedia.org/wiki/Content_management_system). As a rule of thumb, if a client wants a website whose main purpose is to showcase some content, their brand or image, and custom business logic requirements are limited, then chances are that a CMS will fit the bill nicely.
```

Comment:
Perhaps you mean "proprietary, paid, commercial" when you say "off the shelf", but to me that isn't assumed. Many people nowadays say "off the shelf" when they mean an open source app that runs as is, vs. a bunch of open source components that take a bunch of work to assemble and configure. Most readers of our blog expect the things we mention are free software / open source / libre unless we clearly say otherwise.

I think it's important that we mention clearly and early that Craft CMS is proprietary, commercial, not free software or open source. Though as a PHP app, the source is available, and doesn't appear to be obfuscated, I don't see a license or copyright stated anywhere in the downloaded code or on their website, which is really odd! So by default under current copyright law in most of the world, it's proprietary and can't be used without permission.

Let me know if I'm making sense here or not. If this is a pain for you to consider, I'm happy to write some text for you to start with.

Reply:
Gotcha. By "off the shelf" I mean "product that is ready to use as is". I hadn't really considered the open-source-ness of Craft, but sure enough, it seems like this is "open source" more by consequence of the tech used rather than intent. Looking at their license (https://github.com/craftcms/cms/blob/develop/LICENSE.md) and their pricing structure (https://craftcms.com/pricing), only a subset of the features are available for free. Even though the source code is freely available, and their license does give the freedom to "use, copy, modify, merge, publish and/or distribute", they have a number of features hidden behind a paywall.

I've made mention of this in the article, but also feel free to adjust how you see fit of course. It's the second paragraph below this one.